.ci/pipeline/ci.yml

```diff
@@ -57,6 +57,65 @@ jobs:
       pip install isort black[jupyter]==24.1.1
       isort --profile black --check . && black --check .
     displayName: 'Linting'
+- job: LinuxCondaRecipe
+  dependsOn: Lint
+  timeoutInMinutes: 120
+  strategy:
+    matrix:
+      Python3.12_Numpy1.26:
+        PYTHON_VERSION: '3.12'
+        NUMPY_VERSION: '1.26'
+      Python3.9_Numpy1.22:
+        PYTHON_VERSION: '3.9'
+        NUMPY_VERSION: '1.22'
+  pool:
+    vmImage: 'ubuntu-22.04'
+  steps:
+  - script: |
+      bash .ci/scripts/install_dpcpp.sh
+    displayName: "dpcpp installation"
+  - script: |
+      conda config --add channels conda-forge
+      conda config --set channel_priority strict
+      conda update -y -q --all
+    displayName: "Conda update"
+  - script: |
+      conda create -y -q -n build-env conda-build conda-verify
+    displayName: "Conda create"
+  - script: |
+      export DPCPPROOT=/opt/intel/oneapi/compiler/latest
+      . /usr/share/miniconda/etc/profile.d/conda.sh
+      conda activate build-env
+      conda build . --python $(PYTHON_VERSION) --numpy $(NUMPY_VERSION)
+    displayName: "Conda build"
+- job: WindowsCondaRecipe
+  dependsOn: Lint
+  timeoutInMinutes: 120
+  strategy:
+    matrix:
+      Python3.12_Numpy1.26:
+        PYTHON_VERSION: '3.12'
+        NUMPY_VERSION: '1.26'
+      Python3.9_Numpy1.22:
+        PYTHON_VERSION: '3.9'
+        NUMPY_VERSION: '1.22'
```

Comment:
So hardcoding the numpy versions is going to increase maintenance burden, is there a reason why they are pinned, and why pinned to < 2.0?

Reply:
Current numpy handling is similar to `dependencies-dev` and `requirements-test.txt`: lowest possible numpy version is set for building and `{{ pin_compatible(numpy)) }}` installs highest compatible version for testing. The only difference is `conda-build` ignores `numpy>=2.0`